gammapy/stats/feldman_cousins.py

```diff
@@ -0,0 +1,436 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Provide a Feldman Cousins algorithm."""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+import matplotlib.pylab as plt
+from scipy import stats
+
+__all__ = ['find_confidence_interval_gauss',
+           'find_confidence_interval_poisson'
+           'construct_confidence_belt_PDFs',
+           'get_upper_and_lower_limit',
+           'fix_upper_and_lower_limit',
+           'find_limit',
+           'find_average_upper_limit',
+           'construct_confidence_belt']
```

Comment:
Please add the example file that produces the plots to `gammapy/docs/stats/`, e.g. as `plot_feldman_cousins.py` and then include the plot in the docs via a [Sphinx plot directive](http://matplotlib.org/sampledoc/extensions.html#inserting-matplotlib-plots).
Probably all seven functions should be part of the public `gammapy.stats` API then.

We could also ask some stats expert, e.g. Uli Schwanke, to have a look at this PR and ask for advice if he thinks the public API is good, or if something should be added or could be removed.